r/src/array.cpp

```diff
@@ -25,6 +25,8 @@ namespace r {
 
 // the integer64 sentinel
 static const int64_t NA_INT64 = std::numeric_limits<int64_t>::min();
+static const int64_t MAX_INT32 = std::numeric_limits<int32_t>::max();
+static const int64_t MIN_INT32 = std::numeric_limits<int32_t>::min();
```

Comment:
change these to constexpr?

Reply:
Sure. Does that mean they have to be functions ? 

Otherwise, since `::minuend `::max` are themselves `constexpr` perhaps we don't even need the constants here. 

Also, not sure what `MIN_INT32` is used for, but we need to be careful because it's also the `NA` sentinel for `int` in R.

Reply:
yes they are constexpr, e.g. https://github.com/apache/arrow/blob/b165c869e4758b82bd4b25dd307380e22d1a109b/cpp/src/arrow/util/hash.h#L32